README.md

```diff
@@ -15,101 +17,172 @@ This project defines core build rules for [Scala](https://www.scala-lang.org/) t
-* [scala_test](docs/scala_test.md)
-* [scala_repl](docs/scala_repl.md)
-* [scala_library_suite](docs/scala_library_suite.md)
-* [scala_test_suite](docs/scala_test_suite.md)
-* [thrift_library](docs/thrift_library.md)
-* [scala_proto_library](docs/scala_proto_library.md)
-* [scala_toolchain](docs/scala_toolchain.md)
-* [scala_import](docs/scala_import.md)
-* [scala_doc](docs/scala_doc.md)
+- [scala_library](docs/scala_library.md)
+- [scala_macro_library](docs/scala_macro_library.md)
+- [scala_binary](docs/scala_binary.md)
+- [scala_test](docs/scala_test.md)
+- [scala_repl](docs/scala_repl.md)
+- [scala_library_suite](docs/scala_library_suite.md)
+- [scala_test_suite](docs/scala_test_suite.md)
+- [thrift_library](docs/thrift_library.md)
+- [scala_proto_library](docs/scala_proto_library.md)
+- [scala_toolchain](docs/scala_toolchain.md)
+- [scala_import](docs/scala_import.md)
+- [scala_doc](docs/scala_doc.md)
 
 ## Getting started
 
-1. [Install Bazel](https://docs.bazel.build/versions/master/install.html), 
-see the [compatibility table](#bazel-compatible-versions).
-2. Add the following to your `WORKSPACE` file and update versions with their sha256s if needed:
+[Install Bazel][], preferably using the [Bazelisk][] wrapper. See the
+[compatbile Bazel versions](#compatible-bazel-versions) section to select a suitable
+Bazel version.
+
+[Install Bazel]: https://docs.bazel.build/versions/master/install.html
+[Bazelisk]: https://docs.bazel.build/versions/master/install.html
 
-```starlark
+Add the following configuration snippet to your `WORKSPACE` file and update
+versions with their sha256s if needed. This snippet is designed to ensure that
+users pick up the correct order of dependencies for `rules_scala`. If you want
+to override any of the following dependency versions, make sure to `load()` them
+before calling `rules_scala_dependencies()`.
+
+```py
 # WORKSPACE
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
+# See https://github.com/bazelbuild/rules_scala/releases for up to date version
+# information, including `<SHASUM>` values.
 http_archive(
-    name = "bazel_skylib",
-    sha256 = "b8a1527901774180afc798aeb28c4634bdccf19c4d98e7bdd1ce79d1fe9aaad7",
-    urls = [
-        "https://mirror.bazel.build/github.com/bazelbuild/bazel-skylib/releases/download/1.4.1/bazel-skylib-1.4.1.tar.gz",
-        "https://github.com/bazelbuild/bazel-skylib/releases/download/1.4.1/bazel-skylib-1.4.1.tar.gz",
-    ],
+    name = "rules_scala",  # Can be "io_bazel_rules_scala" if you still need it.
+    sha256 = "<SHASUM>",
+    strip_prefix = "rules_scala-7.0.0",
+    url = "https://github.com/bazelbuild/rules_scala/releases/download/7.0.0/rules_scala-7.0.0.tar.gz",
 )
 
-# See https://github.com/bazelbuild/rules_scala/releases for up to date version information.
+load("@rules_scala//scala:deps.bzl", "rules_scala_dependencies")
+
+rules_scala_dependencies()
+
+load("@rules_java//java:rules_java_deps.bzl", "rules_java_dependencies")
```

Comment:
I tried doing this, and got:

```
ERROR: Error computing the main repository mapping: cannot load '@@rules_java//java:rules_java_deps.bzl': no such file
```

Reply:
D'oh! Thanks for that catch. Fixed in https://github.com/bazelbuild/rules_scala/pull/1703/commits/85519d06bbef243b8099ddf76ed6c75e96d8b5ff.

As I mention in that commit message, this was the correct snippet for `rules_java` 8.x, used in my Bazel 8 compatibility branch. It's now updated for `rules_java` 7.x, matching what's in the current `WORKSPACE` files, with some comments explaining what to expect when `rules_java` 8.x lands in this repo.

Reply:
Probably we'd need a documentation change when we'd land `rules_java` 8.x (maybe add that on a branch already?).

Reply:
Of course! I've already been updating the `README.md` in [bzlmod-bazel-8](https://github.com/mbland/rules_scala/tree/bzlmod-bazel-8). 😉 As I've described in several of my "weekly updates" from #1482, I've been keeping that branch and [bzlmod](https://github.com/mbland/rules_scala/tree/bzlmod) up to date with the latest merges to `master` and other developments.

This PR contains the bulk of the necessary doc updates, so it's a much lighter lift to include doc updates in those other branches now.